modules/api/src/main/java/com/intuit/wasabi/api/ApplicationsResource.java

```diff
@@ -373,4 +376,48 @@ public Response getPagesAndAssociatedExperimentsForApplication(
             throw exception;
         }
     }
+
+    /**
+     * Returns a Map of allowed Applications to the corresponding
+     * tags that have been stored with them.
+     *
+     * @param authorizationHeader the authorization headers
+     * @return Response object containing the Applications with their tags
+     */
+    @GET
+    @Path("/tags")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Returns a Map of Applications to their tags",
+            response = Map.class)
+    @Timed
+    public Response getAllExperimentTags(
+            @HeaderParam(AUTHORIZATION)
+            @ApiParam(value = EXAMPLE_AUTHORIZATION_HEADER, required = true)
+            final String authorizationHeader) {
+        try {
+
+            // get the for the user visible Applications
+            List<Application.Name> allApplications = experiments.getApplications();
+            Set<Application.Name> allowed = new HashSet<>();
+            UserInfo.Username userName = authorization.getUser(authorizationHeader);
+
+            for (Application.Name applicationName : allApplications) {
```

Comment:
@AndreaSuckro This loop will end up making a cassandra query for each application (for all the applications in the system irrespective of what the user is authorized to) which in my opinion is highly inefficient. Would suggest using the `getUserPermissionsList()` method instead of `checkUserPermissions()` which would fetch all the user permissions for all applications that it is authorized to view with the application names. You can probably then also remove the `getApplications()` call which will be redundant in that case.